app/javascript/components/App.js

```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import Greeting from './Greeting';
+import { BrowserRouter, Routes, Route } from 'react-router-dom';
+import { Provider } from 'react-redux';
```

Comment:
**OPTIONAL** - You have done well so far, though your project might run well locally on your computer, but there would be errors when it is run on another PC. This is because when compiling the JS for your application, you are using 2 libraries that are not in your project dependencies in your `package.json` file. Please add the `react-router-dom` and `react-redux` to your project dependency 🙏.
![erfgrege](https://github.com/Alibaba2023/hello-rails-react/assets/73607512/d4c815a6-077e-4cec-9a19-1f65a3dbdd6c)


```javascript
   npm install react-router-dom --save
```
```Javascript
   npm install react-redux --save
```